main.go

```diff
@@ -227,7 +227,7 @@ func main() {
 	}
 
 	// Setup Schemes for SSO if template instance is available.
-	if argocd.IsTemplateAPIAvailable() {
+	if argocd.CanUseKeycloakWithTemplate() {
```

Comment:
WDYT about adding a log statement which indicates whether the functionality will be enabled/disabled, based on the results of the call to `InspectCluster` above?